### DISCUSSION TITLE
Behaviour in response to `Delete`-`Actor` and `Undo`-`Accept`-`Follow` activities?

### DISCUSSION THREAD

**Discussion by Tamschi** [snarfed/bridgy-fed#1440] [Q&A] [answered]
I'm not entirely clear on how Bridgy Fed handles these activities currently.

If I'm not mistaken, a `Delete`-`Actor` at most deactivates the bridge account on Bluesky.
It may be good to instead mark the account as deleted to request purging content from the AppView.

It seems Mastodon at least understands `Undo`-`Accept`-`Follow`, [documented here](https://docs.joinmastodon.org/spec/activitypub/#supported-activities-for-profiles).
I think that too should deactivate the bridge (when received targeting the bridge account's `Follow`), since otherwise desyncs may occur in terms of deleted notes being visible on Bluesky, if the author's instance doesn't broadcast their Public activity to Bridgy Fed anymore.

**Comment by snarfed** [comment, marked as answer]
Funny, we've actually moved the opposite way on actor deletes. Up until a week ago, we translated an AP `Delete` of an actor into a Bluesky `#tombstone` of the repo, which couldn't be undone. We switched to `#account active=false` in https://github.com/snarfed/bridgy-fed/issues/1130 to allow it to be undone. I think both flush the repo from the AppView, and re-activating re-fetches it with `getRepo` etc, but afaik those are implementation details, not strictly specified by the protocol.

If someone removes @bsky.brid.gy@bsky.brid.gy as a follower, and their instance sends an `Undo-Accept-Follow`, you're right, we could consider that as also deactivating the bridge. Probably belongs in https://github.com/snarfed/bridgy-fed/issues/1335. I wouldn't expect that to change where `Delete`s for notes etc are sent though, those should be sent everywhere the original `Create` was sent.

**Reply by Tamschi** [reply]
> […], those should be sent everywhere the original `Create` was sent.

At least Mastodon [doesn't track that](https://github.com/mastodon/mastodon/blob/babee06794e188dffb9f82787cc0f0ab784dc99e/app/services/remove_status_service.rb#L89-L100), [no](https://github.com/mastodon/mastodon/blob/babee06794e188dffb9f82787cc0f0ab784dc99e/app/lib/status_reach_finder.rb#L22-L42). I'll add this to #1335.

Regarding deactivation and reactivation, it would be interesting to see if Bluesky indeed tries to re-fetch.
(atproto-browser.vercel.app [fails to fetch collections](https://atproto-browser.vercel.app/at/Tamschi.mastodon.gamedev.place.ap.brid.gy) for bridge accounts, so it's unclear to me whether that would be possible. I _think_ I read somewhere that Bluesky has a timeout even for account deletions during which they are reversible, but I'm not sure if that's implemented in the PDS, AppView or both.)

**Reply by snarfed** [reply]
> At least Mastodon [doesn't track that](https://github.com/mastodon/mastodon/blob/babee06794e188dffb9f82787cc0f0ab784dc99e/app/services/remove_status_service.rb#L89-L100), [no](https://github.com/mastodon/mastodon/blob/babee06794e188dffb9f82787cc0f0ab784dc99e/app/lib/status_reach_finder.rb#L22-L42). I'll add this to https://github.com/snarfed/bridgy-fed/issues/1335.

Sigh, that's disappointing. Thanks for the research.

**Reply by Tamschi** [reply]
For what it's worth, I think no common software tracks per-post reach accurately.
(It'd have to include domains seen in fetch requests too.)

Some may send out `Delete`s more broadly than Mastodon does, though.
In either case, I think it may be helpful to eventually add a "delete [e.g. Bluesky URL or rkey]" command to remove a specific bridged post.
